src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -4,29 +4,111 @@
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import io.stargate.sgv2.jsonapi.api.model.command.clause.update.UpdateClause;
 import io.stargate.sgv2.jsonapi.api.model.command.clause.update.UpdateOperation;
+import io.stargate.sgv2.jsonapi.config.constants.DocumentConstants;
+import io.stargate.sgv2.jsonapi.exception.ErrorCode;
+import io.stargate.sgv2.jsonapi.exception.JsonApiException;
+import io.stargate.sgv2.jsonapi.util.JsonNodeComparator;
 import java.util.List;
 
 /** Updates the document read from the database with the updates came as part of the request. */
-public record DocumentUpdater(List<UpdateOperation> updateOperations) {
+public record DocumentUpdater(
```

Comment:
@tatu-at-datastax I believe extending DocumentUpdater for replace, hope this fine.

Reply:
Yes, makes sense wrt functionality. If we need more types in future may want to refactor.
I'll add one idea down below however.